### PR TITLE
Added new API errors for LDAP

### PR DESCRIPTION
**Pull request by taran-p**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
Adds API errors:
If LDAP is not enabled with an LDAP command
If LDAP short username is expected but DN is given

Uses these in the code

## Motivation and Context
Currently these known errors return as internal errors

## How to test this PR?
`mc idp ldap accesskey create` with LDAP not enabled, with incorrect username, or with a DN
`mc idp ldap policy attach/detach` with LDAP not enabled

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by vadmeste — APPROVED** [review]
LGTM

File: cmd/api-errors.go [hunk 1]
```diff
@@ -278,6 +278,7 @@ const (
 	ErrMalformedJSON
 	ErrAdminNoSuchUser
 	ErrAdminNoSuchUserLDAPWarn
+	ErrAdminLDAPExpectedShortName
```
Comment (donatello) [inline cmd/api-errors.go]: Could you change this to login name? Short name is an internal term.

File: cmd/api-errors.go [hunk 2]
```diff
@@ -2079,7 +2081,16 @@ var errorCodes = errorCodeMap{
...
 	},
-	// Add your error structure here.
+	ErrAdminLDAPNotEnabled: {
+		Code:           "XMinioLDAPNotEnabled",
+		Description:    "LDAP is not enabled. LDAP must be enabled to make LDAP requests.",
+		HTTPStatusCode: http.StatusNotImplemented,
+	},
+	ErrAdminLDAPExpectedShortName: {
+		Code:           "XMinioLDAPExpectedShortName",
+		Description:    "Expected LDAP short username but was given full DN.",
```
Comment (donatello) [inline cmd/api-errors.go:2091]: Here as well. Use Login username

File: cmd/admin-handlers-idp-ldap.go [hunk 3]
```diff
@@ -262,9 +267,13 @@ func (a adminAPIHandlers) AddServiceAccountLDAP(w http.ResponseWriter, r *http.R
 		targetUser, targetGroups, err = globalIAMSys.LDAPConfig.LookupUserDN(targetUser)
 		if err != nil {
 			// if not found, check if DN
-			if strings.Contains(err.Error(), "not found") && globalIAMSys.LDAPConfig.ParsesAsDN(targetUser) {
-				// warn user that DNs are not allowed
-				err = fmt.Errorf("Must use short username to add service account. %w", err)
+			if strings.Contains(err.Error(), "not found") {
```
Comment (donatello) [inline cmd/admin-handlers-idp-ldap.go]: Not found may not be specific enough here. Our ldap wrapper library returns a more specific DN not found or similar, could you use that instead?

**Review by donatello — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by donatello — APPROVED** [review]

**Pull request merged**